homeassistant/components/media_player/firetv.py

```diff
@@ -195,60 +178,132 @@ def source_list(self):
         """Return a list of running apps."""
         return self._running_apps
 
+    @adb_decorator(override_available=True)
     def update(self):
         """Get the latest date and update device state."""
-        self._state = {
-            'idle': STATE_IDLE,
-            'off': STATE_OFF,
-            'play': STATE_PLAYING,
-            'pause': STATE_PAUSED,
-            'standby': STATE_STANDBY,
-            'disconnected': STATE_UNKNOWN,
-        }.get(self._firetv.state, STATE_UNKNOWN)
-
-        if self._state not in [STATE_OFF, STATE_UNKNOWN]:
-            self._running_apps = self._firetv.running_apps
-            self._current_app = self._firetv.current_app
-        else:
+        # Check if device is disconnected.
+        if not self._available:
             self._running_apps = None
             self._current_app = None
 
+            # Try to connect
+            self.firetv.connect()
+            self._available = True
+
+        else:
```

Comment:
If connection has succeeded here, shouldn't we update state? Probably either just remove this `else:` or remove it and `return` in the if block above if we shouldn't update state.

Reply:
Fixed in https://github.com/home-assistant/home-assistant/pull/17767/commits/21d15f28582fd3d8393ed203945f9131969ff8e0 and https://github.com/home-assistant/home-assistant/pull/17767/commits/669946dd7a5b8fba7277f96808c4ba3cfb968915